### PR TITLE
[BugFix]Fix materialized view ttl not remove base partition version

### LINKED ISSUE #14988: [materialized view]result error when set ttl=-1
### Steps to reproduce the behavior (Required)
admin set frontend config("enable_experimental_mv" = "true");
admin set frontend config("dynamic_partition_check_interval_seconds" = "3");
CREATE TABLE t1 (
                            k1 INT,
                            v1 INT,
                            v2 INT)
                        DUPLICATE KEY(k1)
                        PARTITION BY RANGE(`k1`)
                        (
                        PARTITION `p1` VALUES LESS THAN ('2'),
                        PARTITION `p2` VALUES LESS THAN ('3'),
                        PARTITION `p3` VALUES LESS THAN ('4'),
                        PARTITION `p4` VALUES LESS THAN ('5'),
                        PARTITION `p5` VALUES LESS THAN ('6'),
                        PARTITION `p6` VALUES LESS THAN ('7')
                        )
                        DISTRIBUTED BY HASH(k1);
                        insert into t1 values (1,1,1),(1,1,2),(1,2,1),(1,2,2),
                                              (2,1,1),(2,1,2),(2,2,1),(2,2,2),
                                              (3,1,1),(3,1,2),(3,2,1),(3,2,2);
CREATE MATERIALIZED VIEW mv1
               PARTITION BY k1
               DISTRIBUTED BY HASH(k1) BUCKETS 10
               REFRESH ASYNC
               PROPERTIES(
               "partition_ttl_number"="4"
               )
               AS SELECT k1, sum(v1) as sum_v1 FROM t1 group by k1;
select sleep(10);
ALTER MATERIALIZED VIEW mv1 SET ("partition_ttl_number"="-1");
select sleep(10);
select * from mv1;

### Expected behavior (Required)
```
+------+--------+
| k1   | sum_v1 |
+------+--------+
|    2 |      6 |
|    1 |      6 |
|    3 |      6 |
+------+--------+
```
### Real behavior (Required)
```
+------+--------+
| k1   | sum_v1 |
+------+--------+
|    3 |      6 |
+------+--------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.5-RELEASE d026090 |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14988

## Problem Summary(Required) ：

This is because ttl does not take into account the update of metadata. The update of metadata is more complicated for rollup. First, simply fix the bug

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java [hunk 1]
```diff
@@ -389,4 +396,41 @@ public static Range<PartitionKey> createRange(String lowerBound, String upperBou
...
+    public static void dropBaseVersionMeta(MaterializedView mv, String basePartitionName) {
+        MaterializedView.AsyncRefreshContext refreshContext = mv.getRefreshScheme().getAsyncRefreshContext();
+        Map<Long, Map<String, MaterializedView.BasePartitionInfo>> versionMap =
+                refreshContext.getBaseTableVisibleVersionMap();
+        if (versionMap == null) {
+            return;
+        }
+        Expr expr = mv.getPartitionRefTableExprs().get(0);
+        SlotRef slotRef;
+        if (expr instanceof SlotRef) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java:409]: keep the same with line 426? I think they are the same?
add some comment for this condition

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 26 / 30 (86.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|[340]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|23|26|88.46%|[404, 418, 422]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**